### ISSUE TITLE
ConnectionLost is documented in twisted.words.protocols.jabber.xmlstream instead of twisted.internet.error

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4856]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4856|
|Type|defect|
|Created|2011-02-09 13:23:17Z|
|Branch|https://github.com/twisted/twisted/tree/connectionlost-location-4856|

`ConnectionLost` is in the `__all__` of `twisted.words.jabber.xmlstream` for some reason.  This convinces pydoctor to document it in that module instead of in `twisted.internet.error`.

It should be removed from the `__all__` of `xmlstream` and `twisted.internet.error` should gain an `__all__` including it and the other public attributes of that module.

Attachments:

* [4856.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/989/989583e972d152adc8b275f72e0ef34e5de71a1a/a4da115289afa396f61a38294fe47e9266a34970.patch) (5912 bytes) - added by awclin on 2011-02-19 14:04:02Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4856 4856
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_connectionlost_location_4856 branches/connectionlost-location-4856
branch_author__awclin awclin
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1297257797000000 1297257797000000
changetime__1299025914000000 1299025914000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="awclin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|awclin commented|
|-|-|

I've added twisted.internet.error.__all__, and removed ConnectionLost from twisted.words.protocols.jabber.xmlstream.__all__, and fixed up some spacing/formatting.

 I've manually generated the API docs, and everything looks to be in the right place.  I grep'd around the codebase and couldn't find any tests for a module's __all__ property, nor could I think of a useful way to test it myself (the list contains properties of the module that the implementor _meant_ to make public, and there is no useful syntactic identification in the module contents to convey that meaning).  Let me know if there are any tests you would like adding, and some idea on what they should be.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [30866]) Branching to 'connectionlost-location-4856'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [30867]) Apply patch fixing two `__all__` module attributes

refs [#4856](https://github.com/twisted/twisted/issues/4856)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [30868]) Adjust indentation

refs [#4856](https://github.com/twisted/twisted/issues/4856)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

Thanks.  I tweaked the indentation and the news fragment a little.  Otherwise looks great, I'll apply this now.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [30870]) Merge connectionlost-location-4856

Author: awclin
Reviewer: exarkun
Fixes: [#4856](https://github.com/twisted/twisted/issues/4856)

Relocate `ConnectionLost` into `twisted.internet.error` by giving that module
an `__all__` which includes that name.  Make the corresponding removal from
`twisted.words.protocols.jabber.xmlstream.__all__` which incorrectly included
that exception class's name.

This corrects the pydoctor-generator documentation as well as just correctly
reflecting which module exposes this exception.
